libsplinter/src/circuit/template/yaml_parser/mod.rs

```diff
@@ -0,0 +1,134 @@
+// Copyright 2018-2020 Cargill Incorporated
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+pub mod v0_1;
+
+use std::fs::File;
+use std::io::Read;
+
+use super::Error;
+
+#[derive(Deserialize, Debug)]
+struct TemplateVersionGuard {
+    version: String,
+}
+
+pub fn load_template(file_path: &str) -> Result<Template, Error> {
```

Comment:
Before attempting to open the file, this function could do some basic checks (does the file exist, is the file readable, etc.) to avoid the chained error that results in File::open().

Reply:
Addressed